shoop/configuration.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+
+from shoop.core import cache
+from shoop.core.models import Shop, Configuration
+
+
+SHOP_CONFIGURATION_CACHE_PREFIX = "shop_configuration_%s"
+
+
+def get_cache_key(shop):
+    return SHOP_CONFIGURATION_CACHE_PREFIX % shop.pk if shop else ""
+
+
+def set(shop, key, data):
+    """
+    Set shop, key, data combination to configurations.
```

Comment:
`data` should be renamed to `value`.

The docstring could be something like
```python
def set(shop, key, value):
    """
    Set configuration item value for a shop or globally.

    If given `shop` is ``None``, the value of given `key` is set
    globally for all shops.  Otherwise sets a shop specific value which
    overrides the global value in configuration of the specified shop.

    :type shop: shoop.core.models.Shop|None
    :param shop: Shop to set value for
    :type key: str
    :param key: Name of the key to set
    :type value: Any
    :param value: Value to set.  Note: Must be JSON serializable.
    """
```